### ISSUE TITLE
Closing a plot window throws an error when in pause

### ISSUE THREAD

**Issue by Yazington** [matplotlib/matplotlib#16667]
If I'm using plot.pause(15) and I close the window while there is a pause, the window shouldn't throw an exception, instead keep the execution going

### Bug report

**Bug summary**

**Code for reproduction**

```python
# Paste your code here
#   plt.show(block=False)
    plt.pause(15)
    self.hide()
    def hide(self):
        if plt.gcf():
            plt.clf()
            plt.cla()
            plt.close()
#
```

**Actual outcome**

```
# If applicable, paste the console output here
#_tkinter.TclError: can't invoke "update" command: application has been destroyed
#
```

**Expected outcome**

**Matplotlib version**

  * Operating system:
  * Matplotlib version: 
  * Matplotlib backend (`print(matplotlib.get_backend())`):
  * Python version:
  * Jupyter version (if applicable):
  * Other libraries:

**Comment by tacaswell** [comment]
Can you please fill out the information about Matplotlib, python, system, and tk version?

This is clearly a snippet from a much larger code base, do you have a self-contained example that demonstrates this problem?

**Comment by WeatherGod** [comment]
It would also be useful to see the traceback

**Comment by anntzer** [comment]
closing for lack of activity.

**Comment by tacaswell** [comment]
I suspect that this will be fixed by #17802 .
